### PR TITLE
feat: new message and map question templates

### PR DESCRIPTION
**Pull request by artlee06**
- maps the monaco language names to enable syntax highlighting
- omits the languages supported by leetcode but not supported by monaco
- Gives a chat indicator when a new message comes in

### CODE REVIEW

File: frontend/src/components/Chat.tsx [hunk 1]
```diff
@@ -60,11 +61,27 @@ const Chat: React.FC = function () {
...
   const [content, setContent] = useState<string>('');
+  const isChatVisible = useAppSelector(getIsVisible);
+  const dispatch = useAppDispatch();
 
   const currentUser = firebaseApp.auth().currentUser;
   const uid = currentUser?.uid;
   const displayName = currentUser?.displayName;
 
+  useEffect(() => {
+    if (messages.length == 0) {
```
Comment (ivantjh) [inline frontend/src/components/Chat.tsx:72]: iirc, should use `if (messages.length === 0)`

File: frontend/src/components/Chat.tsx [hunk 2]
```diff
@@ -60,11 +61,27 @@ const Chat: React.FC = function () {
...
+      return;
+    }
+
+    const latestMessage = messages[messages.length - 1];
+    const isFromOtherParty = latestMessage.uid != uid;
```
Comment (ivantjh) [inline frontend/src/components/Chat.tsx:77]: Same for here also `!==`

**Review by ivantjh — APPROVED** [review]
lgtm!

**Comment by artlee06** [comment]
@chesterhow is this to merge for this?

**Review by chesterhow — APPROVED** [review]
LGTM, thanks Arthur!

**Pull request merged**